### PR TITLE
refactor: rework target practice

### PR DESCRIPTION
**Pull request by LordMidas**
- Require bow
- Remove ability to swap ammo for free
- Change bonus to affect enemies not in cover instead of enemies with no adjacent ally

### CODE REVIEW

File: scripts/skills/perks/perk_rf_target_practice.nut [hunk 1]
```diff
@@ -16,40 +16,38 @@ this.perk_rf_target_practice <- ::inherit("scripts/skills/skill", {
...
+		if (!_skill.isRanged() || !_skill.m.IsWeaponSkill || !this.isEnabled())
+			return;
+
 		local targetEntity = _targetTile.getEntity();
-		if (_skill.isRanged() && targetEntity != null && (targetEntity.isArmedWithRangedWeapon() || ::Tactical.Entities.getAlliedActors(targetEntity.getFaction(), targetEntity.getTile(), 1, true).len() == 0))
+		if (targetEntity != null && (targetEntity.isArmedWithRangedWeapon() || ::Const.Tactical.Common.getBlockedTiles(this.getContainer().getActor().getTile(), targetEntity.getTile(), this.getContainer().getActor().getFaction()).len() == 0))
 		{
 			_tooltip.push({
 				icon = "ui/tooltips/positive.png",
 				text = "[color=" + ::Const.UI.Color.PositiveValue + "]" + this.m.RangedSkillBonus + "%[/color] " + this.getName()
```
Comment (Darxo) [inline scripts/skills/perks/perk_rf_target_practice.nut:49]: Might aswell improve the tooltip coloring if you are already working on this perk
`text = ::MSU.Text.colorGreen(this.m.RangedSkillBonus + "%") + this.getName()`

File: scripts/skills/perks/perk_rf_offhand_training.nut [hunk 2]
```diff
@@ -104,10 +104,7 @@ this.perk_rf_offhand_training <- ::inherit("scripts/skills/skill", {
 
 	function onPayForItemAction(_skill, _items)
 	{
-		if (_skill != null && _skill.getID() != "perk.rf_target_practice")
-		{
-			this.m.IsSpent = true;
-		}
+		this.m.IsSpent = true;
```
Comment (Darxo) [inline scripts/skills/perks/perk_rf_offhand_training.nut:107]: This has nothing to do with this PR and should be moved to another PR

**Review by Darxo — CHANGES_REQUESTED** [review]
The skill implementation itself looks good.
Just two minor things around this PR

File: scripts/skills/perks/perk_rf_target_practice.nut (hunk 1 above)
Comment (LordMidas) [reply scripts/skills/perks/perk_rf_target_practice.nut:49]: I avoided doing these changes in general in this perk rework so as not to muddle the actual perk changes with extra lines of just refactoring the tooltip. I'd rather do these changes separately in a single commit in the entire codebase.

File: scripts/skills/perks/perk_rf_offhand_training.nut (hunk 2 above)
Comment (LordMidas) [reply scripts/skills/perks/perk_rf_offhand_training.nut:107]: This perk used to be able to allow swapping ammo for free. This was then accounted for in `rf_offhand_training` where if you are swapping ammo with this perk then that perk didn't become "spent". Now that we removed this ability from the target practice perk, we have to remove that accounting for it in the other perk too. So it is part of this change.

**Review by Darxo — APPROVED** [review]

**Comment by Uberbagel** [comment]
Tested. Working. Log clean.
